src/olympia/users/models.py

```diff
@@ -864,13 +873,21 @@ def allow_request(cls, request):
             request.user.email))
 
 
-class DeveloperAgreementRestriction:
+class DeveloperAgreementRestriction(GetErrorMessageMixin):
     error_message = _('Before starting, please read and accept our Firefox'
                       ' Add-on Distribution Agreement as well as our Review'
                       ' Policies and Rules. The Firefox Add-on Distribution'
                       ' Agreement also links to our Privacy Notice which'
                       ' explains how we handle your information.')
 
+    @classmethod
+    def get_error_message(cls, is_api):
+        if is_api:
+            url = 'https://addons.mozilla.org/en-US/developers/addon/api/key/'
+            return ('{} Please see {}'.format(cls.error_message, url))
```

Comment:
This should be a translated string as well, so it needs to be wrapped in `_()`. But more importantly, thinking about this more, I don't think we need the full message for the API. Developers should only stumble upon this message if they have already accepted the developer agreement but the document has changed since they agreed to it, and they'll see the full error message when they go to the URL.

So, I propose not re-using the message, and doing this:
```suggestion
            return _('Please read and accept our Firefox on Distribution '
                     'Agreement as well as our Review Policies and Rules '
                     'by visiting {url}'.format(url=url))
```